Feat/add version 8 to 9 converter

TheGreatRefrigerator opened the pull request.

merydian committed:
refactor: migrate function name -> migrate_7_to_8

merydian committed:
feat: implement migrate_8_to_9

merydian committed:
test: add test for migrate_8_9()

TheGreatRefrigerator committed:
refactor: function, file and class names

Co-authored-by: merydian <till.frankenbach@gmail.com>

merydian committed:
feat: add v9 models

merydian committed:
feat: add version number arguments to switch between migration functions

merydian committed:
refactor: extract function for warning/error output

merydian committed:
fix: remove {} from produced yaml

TheGreatRefrigerator committed:
fix: add missing astar execution method entry

merydian committed:
fix: link to logging documentation

TheGreatRefrigerator commented:
Currently, graphs_root_path is just removed without further info.
The final output should contain a bit more of what's new and where to find docu for it

TheGreatRefrigerator committed:
docs: add version number arguments to commands

jhaeu commented:
From 8 -> 9 please migrate 

`ors.engine.graphs_root_path` -> `ors.engine.profile_default.graph_path`

merydian commented on migrate.py:
```diff
@@ -412,6 +415,131 @@ def migrate(json_config_path, yaml_config_path):
+        if_exists_move_to(x, f'ors.{level_string}.source_file',
+                          f'ors.{level_string}.profile_default.build.source_file')
+
+        if_exists_move_to(x, f'ors.{level_string}.graphs_root_path', f'ors.{level_string}.profile_default.graphs_root_path')
```
This is the only occurrence

jhaeu replied:
Sure? Is there no model for v9 that needs to be adapted?

jhaeu commented:
I found out that when migrating v7 -> v8 `graphs_root_path` gets lost. This seems to be an old issue.

This peace of information should be located in

* v7 `ors.services.routing.profiles.default_params.graphs_root_path`
* v8 `ors.engine.graphs_root_path`
* v9 `ors.engine.profile_default.graph_path`

TheGreatRefrigerator committed:
fix: 7 to 8 migration

graphs_root_path was not migrated as the top level object was
accessed for moving it, which didn't contain profiles anymore.

maximum_segment_distance_with_dynamic_weights is just removed as
it was an invalid key in 7.0 already

TheGreatRefrigerator commented:
restructured and fixed missing things in 9 allow extras model
